runtime/moonbase/src/xcm_config.rs

```diff
@@ -438,6 +439,28 @@ where
 	}
 }
 
+#[cfg(feature = "runtime-benchmarks")]
+impl From<MultiLocation> for CurrencyId {
+	fn from(location: MultiLocation) -> CurrencyId {
+		use xcm_executor::traits::Convert as XConvert;
+		use xcm_primitives::AssetTypeGetter;
+
+		// If it does not exist, for benchmarking purposes, we create the association
+		let asset_id = if let Ok(asset_id) =
+			AsAssetType::<AssetId, AssetType, AssetManager>::convert_ref(&location)
+		{
+			asset_id
+		} else {
+			let asset_type = AssetType::Xcm(location);
+			let asset_id: AssetId = asset_type.clone().into();
+			AssetManager::set_asset_type_asset_id(asset_type, asset_id);
```

Comment:
Having side effects in a `From` seems very weird (even if it's feature-gated), do you think this will go away with further refactoring of XCM benchmarks?

Reply:
I dont think this part can go away. We are generic over CurrencyId in the xcm-transactor, but we need a way in the benchmarks to compute a currencyId from a MultiLocation. Even further, to properly benchmark, we need the association CurrencyId<-> MultiLocation to exist, otherwise it will early fail. This is the less dirty solution I could came up with..

Reply:
You could also add a warning comment to draw attention to the side effect.

Reply:
Added my suggestions in 0c8b2b8

Reply:
I am tempted to move this into our runtime/common benchmark API. What do you think @notlesh ?

Reply:
Maybe this can be done in the future, as I think it wont be the only time we have to circle back to this.